src/algorithm/coords_iter.rs

```diff
@@ -0,0 +1,89 @@
+use num_traits::Float;
+use std::iter;
+use types::{Coordinate, Point, Polygon, LineString, Line, MultiPoint, MultiPolygon,
+            MultiLineString};
+
+pub trait CoordsIter<T: Float> {
+    fn coords_iter<'a>(&'a self) -> Box<Iterator<Item = &'a Coordinate<T>> + 'a>;
```

Comment:
just like the other boxed iterators that get returned in this crate, we can remove the `Box` once return `impl`s stabilize